library/aes.c

```diff
@@ -1082,16 +1085,20 @@ int mbedtls_aes_crypt_cbc( mbedtls_aes_context *ctx,
             for( i = 0; i < 16; i++ )
                 output[i] = (unsigned char)( input[i] ^ iv[i] );
 
-            mbedtls_aes_crypt_ecb( ctx, mode, output, output );
+            ret = mbedtls_aes_crypt_ecb( ctx, mode, output, output );
+            if( ret != 0 )
+                goto exit;
             memcpy( iv, output, 16 );
 
             input  += 16;
             output += 16;
             length -= 16;
         }
     }
+    ret = 0;
 
-    return( 0 );
+exit:
```

Comment:
I notice that the previous PR #1096 had a call to `mbedtls_platform_zeroize()` here to clear the output buffer when `ret != 0`. Is this needed here?

Reply:
Yes, but adding missing zeroizations is out of scope here. This is deliberate because previous attempts got bogged down in deciding where to add zeroization.